### PR TITLE
Add README for github workflows directory

### LINKED ISSUE #738: Add README to the workflows directory
The `.github/workflows` directory has become pretty full, and will only keep growing! With #728, it will additionally contain one (for now..) non-workflow file that is consumed by a workflow. 
It's time for docs! This PR tracks adding a README to the directory to help orient viewers/contributors/future us to all the files.

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #738 

This PR adds documentation for the files in `.github/workflows/`. Any feedback welcome, including whether a different readme format (e.g. a table instead of a list?) would be preferred.

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+## Building and pushing Docker images
+
+* Each workflow `docker_{module name}.yml` is used to build the given module's Docker image.
+* `docker_all-modules.yml` builds and pushes all module Docker images by calling each `docker_{module name}.yml` workflow.
+* The `build-push-docker-module.yml` workflow is not standlone but contains actions used by `docker_all-modules.yml` and each `docker_{module name}.yml`.
+
+## Module testing
+
+* Each workflow `run_{module name}.yml` is used to run the given module in CI using the test data.
+* `run_all-modules.yml` runs all modules in CI by calling each `run_{module name}.yml` workflow.
```
Comment (jashapiro) [inline .github/workflows/README.md:12]: I would put this first, as we want to do this for all modules, but Docker comes later.

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+This directory contains GitHub Actions (GHA) yaml files as well as helper files they use.
+
+## Building and pushing Docker images
+
+* Each workflow `docker_{module name}.yml` is used to build the given module's Docker image.
+* `docker_all-modules.yml` builds and pushes all module Docker images by calling each `docker_{module name}.yml` workflow.
+* The `build-push-docker-module.yml` workflow is not standlone but contains actions used by `docker_all-modules.yml` and each `docker_{module name}.yml`.
```
Comment (jashapiro) [inline .github/workflows/README.md:7]: Start with the positive!
```suggestion
* The `build-push-docker-module.yml` workflow contains actions used by `docker_all-modules.yml` and each `docker_{module name}.yml`. It is not meant to be run as a standalone action.
```

**Review by jashapiro — APPROVED** [review]
LGTM with a few minor changes.

**Comment by sjspielman** [comment]
Noting I canceled the `cell-type-ewings` Docker workflow. Since there wasn't actually a code change here, this action doesn't need to run here.

**Pull request merged**